### PR TITLE
Handle pom <parent> tag

### PR DESCRIPTION
**Pull request by arichiardi**
* adds support for the `<parent>` tag in `boot.pom` for both writing and reading `pom.xml` files.
* also removes some obsolete file and hooks up boot/worker tests in Makefile.

### CODE REVIEW

File: boot/core/src/boot/task/built_in.clj [hunk 1]
```diff
@@ -504,7 +505,8 @@
                :dependencies deps
                :developers developers
                :classifier classifier
-               :packaging (or packaging "jar"))]
+               :packaging (or packaging "jar")
+               :parent (-> parent vec first))]
```
Comment (arichiardi) [inline boot/core/src/boot/task/built_in.clj]: @micha I wanted to have `a parent SYM:VER=PATH       [sym str str]` like in #578, can I do it?

File: boot/core/src/boot/task/built_in.clj [hunk 2]
```diff
@@ -493,7 +493,8 @@
    s scm KEY=VAL           {kw str}    "The project scm map (KEY is one of url, tag, connection, developerConnection)."
    l license NAME:URL      {str str}   "The map {name url} of project licenses."
    o developers NAME:EMAIL {str str}   "The map {name email} of project developers."
-   D dependencies SYM:VER  [[sym str]] "The project dependencies vector (overrides boot env dependencies)."]
+   D dependencies SYM:VER  [[sym str]] "The project dependencies vector (overrides boot env dependencies)."
+   a parent SYM:VER        {sym str}   "The project dependency vector of the parent project"]
```
Comment (arichiardi) [inline boot/core/src/boot/task/built_in.clj]: @micha I wanted to have `a parent SYM:VER=PATH       [sym str str]` like in #578, can I do it?

**Comment by arichiardi** [comment]
I don't remember what I wanted to do in the above comment. It looks like I wanted to split the argument in three (regression aside) but really don't remember what the `path` was for. In any case the regression should in #578 should probably be addressed first.

**Comment by martinklepsch** [comment]
Is the basic feature here still wanted? Is that task option type stuff a blocker?

Lets go ahead and merge this if not. I'll write a changelog entry after merging.

**Comment by arichiardi** [comment]
I think this makes things nicer when writing actual Maven poms for plugins. I have needed this and it does not hurt.
I need to actually check the bug though.

File: boot/core/src/boot/task/built_in.clj [hunk 3]
```diff
@@ -541,16 +541,17 @@
...
+   v version VER           str           "The project version."
+   d description DESC      str           "The project description."
+   c classifier STR        str           "The project classifier."
+   P packaging STR         str           "The project packaging type, i.e. war, pom"
+   u url URL               str           "The project homepage url."
+   s scm KEY=VAL           {kw str}      "The project scm map (KEY is one of url, tag, connection, developerConnection)."
+   l license NAME:URL      {str str}     "The map {name url} of project licenses."
+   o developers NAME:EMAIL {str str}     "The map {name email} of project developers."
+   D dependencies SYM:VER  [[sym str]]   "The project dependencies vector (overrides boot env dependencies)."
+   a parent SYM:VER=PATH   [sym str str] "The project dependency vector of the parent project, path included."]
```
Comment (arichiardi) [inline boot/core/src/boot/task/built_in.clj]: Only this line changed

**Comment by martinklepsch** [comment]
Yikes I think I might have somehow wrecked this PR by using Githubs web interface to address the conflicts in Changes.md.

Merged now though
- Change: https://github.com/boot-clj/boot/commit/e7898a99c9ff7c94f35af7b2fa6a2df997b1f3bf
- Tests: https://github.com/boot-clj/boot/commit/94696f66945b640b4d26f45b699342db55786938
- Changelog: https://github.com/boot-clj/boot/commit/49ea7d6207e694f98375f57776846732e3c91908

**Pull request closed without merging**